src/api/teams.iced

```diff
@@ -59,6 +59,23 @@ teams.get_all_teams = (callback) ->
   out = _.flatten(resps, true)
   return callback(null, out)
 
+teams.getTeamRolesForUser = (user, callback) ->
+  ###
+  return an array of team/role hashes to which the user belongs:
+    [{team: <obj>, role: <str>}]
+  ###
+  await teams.get_all_teams(defer(err, all_teams))
+  if err then return callback(err)
+
+  team_roles = []
+
+  for team in all_teams
```

Comment:
This may overcomplicate things but perhaps there is a way to store references to the linked roles object by username in the user object on write / remove for easy reference later without the loop? I can imagine this being relatively slow if teams / roles gets built out. Not a must-do, but something that caught my eye if you're routinely doing user-based searches as well.